src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.service.impl;
+
+import java.util.List;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        var shoppingCart = shoppingCartDao.getByUser(user).orElseThrow(()
```

Comment:
```suggestion
        var shoppingCart = getByUser(user);
```
You already have getByUser in service